Mathlib/Algebra/Category/Module/Projective.lean

```diff
@@ -0,0 +1,70 @@
+/-
+Copyright (c) 2020 Markus Himmel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Markus Himmel, Scott Morrison
+
+! This file was ported from Lean 3 source module algebra.category.Module.projective
+! leanprover-community/mathlib commit 201a3f4a0e59b5f836fe8a6c1a462ee674327211
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Algebra.Category.ModuleCat.EpiMono
+import Mathlib.Algebra.Module.Projective
+import Mathlib.CategoryTheory.Preadditive.Projective
+import Mathlib.LinearAlgebra.FinsuppVectorSpace
+
+/-!
+# The category of `R`-modules has enough projectives.
+-/
+
+
+universe v u
+
+open CategoryTheory
+
+open CategoryTheory.Limits
+
+open LinearMap
+
+open Module
+
+/-- The categorical notion of projective object agrees with the explicit module-theoretic notion. -/
+theorem IsProjective.iff_projective {R : Type u} [Ring R] {P : Type max u v} [AddCommGroup P]
+    [Module R P] : Module.Projective R P ↔ Projective (ModuleCat.of R P) := by
+  refine' ⟨fun h => _, fun h => _⟩
+  · letI : Module.Projective R (ModuleCat.of R P) := h
+    exact ⟨fun E X epi => Module.projective_lifting_property _ _ ((ModuleCat.epi_iff_surjective _).mp epi)⟩
+  · refine' Module.Projective.of_lifting_property.{u,v} _
+    intro E X mE mX sE sX f g s
+    haveI : Epi (ModuleCat.asHom f) := (ModuleCat.epi_iff_surjective (ModuleCat.asHom f)).mpr s
```

Comment:
For some reason the operator `↟` is not working.

Reply:
I can't find `↟` in the source? If you delete stuff from the mathport output, please put in comments showing the original code!

Reply:
`open ModuleCat`